File: runme/mcgen/local/datan/datsrc/mtxgsv.f
      SUBROUTINE MTXGSV(U,V,N,NRED,LIST)
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      DIMENSION U(N),V(NRED),LIST(N)
      IRED=0
      DO 10 I=1,N
        IF(LIST(I).EQ.1) THEN
          IRED=IRED+1
          V(IRED)=U(I)
        END IF
   10 CONTINUE
      END

File: runme/mcgen/local/datan/datsrc/mtxgsv.c
#include "datsrc.h"

void LIBFUNC mtxgsv_(double const P_T *F_P(u), double P_T *F_P(v),
                    integer const P_T *n, integer const P_T *nred,
                    integer const P_T *list)
{
    /* System generated locals */
    integer i__1;

    /* Local variables */
    integer ired, i;

#if defined(USEHUGE)
    HFP(double,u); HFP(double,v);
#endif

    /* Parameter adjustments */
    --list;
    --u;
    --v;

    /* Function Body */
    ired = 0;
    i__1 = *n;
    for (i = 1; i <= i__1; ++i) {
        if (list[i] == 1) {
            ++ired;
            v[ired] = u[i];
        }
    }
    return;
} /* mtxgsv_ */

